lib/index.js

```diff
@@ -64,30 +69,41 @@ _.assign(PostmanSandbox.prototype, {
         options.debug = this.debug;
 
         var id = uuid(),
-            executionEventName = `execution.${id}`;
-        this.executing[id] = true;
+            executionEventName = `execution.${id}`,
+            executionTimeout = _.get(options, 'timeout', this.executionTimeout);
+
+        // set execution timeout and store the interrupt in a global object (so that we can clear during dispose)
+        // force trigger of the `execution.${id}` event so that the normal error flow is taken
+        this.executing[id] = _.isFinite(executionTimeout) ? (setTimeout(this.emit.bind(this, executionEventName,
+            new Error(EXECUTION_TIMEOUT_ERROR_MESSAGE)), executionTimeout)) : null;
 
         // @todo decide how the results will return in a more managed fashion
+        // listen to this once, so that subsequent calls are simply dropped. especially during timeout and other
+        // errors
         this.once(executionEventName, function (err, result) {
-            delete this.executing[id];
+            if (this.executing.hasOwnProperty(id)) { // clear any pending timeouts
+                this.executing[id] && clearTimeout(this.executing[id]);
+                delete this.executing[id];
+            }
+
             this.emit('execution', err, id, result);
             callback(err, result);
         });
 
-        // set execution timeout
-        // @todo add tests
-        _.isFinite(options.timeout) && setTimeout(this.emit.bind(this, executionEventName,
-            new Error('sandbox: execution timeout')), options.timeout);
-
         // send the code to the sendbox to be intercepted and executed
         this.dispatch('execute', id, code, options);
     },
 
     dispose: function () {
-        Object.keys(this.executing).forEach(function (id) {
-            this.emit(`execution.${id}`, new Error('sandbox: execution interrupted, bridge disconnecting.'));
+        var self = this;
+
+        _.forEach(self.executing, function (irq, id) {
+            irq && clearTimeout(irq);
+            self.emit(`execution.${id}`, new Error(BRIDGE_DISCONNECTING_ERROR_MESSAGE));
```

Comment:
fixed a bug here where `this` was not referring to correct item. Shows that we never disposed sandbox in any tests or code!! ;-)